sklearn/neighbors/_base.py

```diff
@@ -1108,12 +1119,17 @@ class from an array representing our data set and ask who's
             else:
                 parallel_kwargs = {"prefer": "threads"}
 
-            chunked_results = Parallel(n_jobs, **parallel_kwargs)(
-                delayed_query(
-                    self._tree, X[s], radius, return_distance, sort_results=sort_results
+            with config_context(assume_finite=True):
+                # We remove the validation of the query points
+                # (in *parallel_kwargs) done at the beginning of
+                # BinaryTree.query_radius as those points already
+                # got validated in the caller.
```

Comment:
Checks running time here is negligible here similarly.
<details>

<summary> Script </summary>

```python
# profile.py
from sklearn.datasets import make_regression
from sklearn.neighbors import NearestNeighbors


def main(args=None):
    X_train, _ = make_regression(n_samples=100_000, n_features=10)
    X_test, _ = make_regression(n_samples=100_000, n_features=10)

    nn = NearestNeighbors(algorithm='kd_tree').fit(X_train)
    nn.kneighbors(X_test, n_neighbors=2)


if __name__ == "__main__":
    main()

```

```sh
giltracer --state-detect profile.py
```

</details>